src/content/blog/starlight-032.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Starlight 0.32
+description: The latest Starlight features unlock new possibilities on our path to v1. Let’s meet route middleware, plugin i18n APIs, and multisite search.
+publishDate: 2025-02-17
+authors:
+  - chris
+coverImage: '/src/content/blog/_images/starlight-032/blog-post-starlight-v0.32.webp'
+socialImage: '/src/content/blog/_images/starlight-032/og-starlight-v0.32.webp'
+---
+
+**Starlight 0.32 is here! Let’s unpack what’s new as we work our way towards a v1 release.**
+
+🧘 We’ve been stretching to make Starlight extra extendable. Here’s our latest flow for more flexibility:
+
+- [**Route data middleware**](#a-new-route-data-paradigm)
+- [**New i18n APIs for plugins**](#new-i18n-apis-for-plugins)
+- [**Updated plugin hooks**](#updated-plugin-hooks)
+- [**Multisite search support**](#multisite-search-support)
+
+To upgrade an existing Starlight site, use the automated `@astrojs/upgrade` CLI tool. This will update Starlight, Astro, and any other integrations you are using:
+
+```sh
+npx @astrojs/upgrade
+```
+
+## A new route data paradigm
+
+Historically, users and plugin authors have found themselves reaching for Starlight’s [component override](https://starlight.astro.build/guides/overriding-components/) system when all they really wanted to do was tweak the data rendered by Starlight’s default components.
+
+In v0.32, we are introducing **a new route middleware system** that gives users full access to Starlight’s data model without needing to override any components. This is particularly helpful for plugins as it avoids common conflicts where several plugins tried to override the same component.
+
+Similar to Astro’s middleware system, route middleware is called for every Starlight page render and gives you the opportunity to modify any data before it gets rendered. This is a powerful way to implement custom logic not currently possible with configuration alone.
```

Comment:
```suggestion
Similar to Astro’s middleware system, a route middleware is called for every Starlight page render and gives you the opportunity to modify any data before it gets rendered. This is a powerful way to implement custom logic not currently possible with configuration alone.
```

Is this missing "a" or maybe "route middlewares are"? Or maybe it's fine like this, just sounded weird to me.

Reply:
Very good question 😅

I… actually have no idea now you ask this. For some reason in my head “middleware” sounds like a plural/collective noun, e.g. often people will say things like “this website uses middleware” rather than “a middleware” or “middlewares”. So here it is describing the full stack of middleware being called not a specific one. 

Looking around I think this aligns with its related words like “software” which also doesn’t usually get pluralized to “softwares”. The [Wiktionary definition](https://en.wiktionary.org/wiki/middleware) also describes it as “usually uncountable”, which I guess means the same thing?